### DISCUSSION TITLE
single-spa integration

### DISCUSSION THREAD

**Discussion by nrgapple** [trpc/trpc#1988] [General]
We use single-spa orchestrators and I was wondering if anyone has experimented with a template to use trpc in a single-spa

https://single-spa.js.org/docs/microfrontends-concept

**Comment by jlalmes** [comment]
Take a read of this: https://adostes.medium.com/reusing-react-query-across-micro-frontends-with-single-spa-and-webpack-module-federation-bf161bd4989e
> In this article, I will show how to setup multiple micro frontends using [single-spa](https://single-spa.js.org/) and [module federation](https://webpack.js.org/concepts/module-federation/). Each micro frontend will be using [React Query](https://react-query.tanstack.com/) to fetch and cache data. I will then demonstrate how multiple micro frontends can use the same [QueryClient](https://react-query.tanstack.com/reference/QueryClient) to share cached data

`@trpc/client` uses `react-query` under the hood, so this should be entirely possible!

**Comment by chriba** [comment]
Hey, this looks really interesting. Has anyone already looked into this in more detail?
